msgraph-mail/typescript/src/models/attachmentCollectionResponseImpl.ts

```diff
@@ -0,0 +1,47 @@
+import {Attachment} from './attachment';
+import {AttachmentCollectionResponse} from './attachmentCollectionResponse';
+import {createAttachmentFromDiscriminatorValue} from './createAttachmentFromDiscriminatorValue';
+import {AttachmentImpl} from './index';
+import {AdditionalDataHolder, Parsable, ParseNode, SerializationWriter} from '@microsoft/kiota-abstractions';
+
+export class AttachmentCollectionResponseImpl implements AttachmentCollectionResponse {
+    /** Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well. */
+    public additionalData: Record<string, unknown>;
+    /** The nextLink property */
+    public nextLink?: string | undefined;
+    /** The value property */
+    public value?: Attachment[] | undefined;
+    /**
+     * Instantiates a new AttachmentCollectionResponse and sets the default values.
+     * @param attachmentCollectionResponseParameterValue 
+     */
+    public constructor(attachmentCollectionResponseParameterValue?: AttachmentCollectionResponse | undefined) {
+        this.additionalData = attachmentCollectionResponseParameterValue?.additionalData ? attachmentCollectionResponseParameterValue?.additionalData! : {};
+        this.nextLink = attachmentCollectionResponseParameterValue?.nextLink;
+        this.value = attachmentCollectionResponseParameterValue?.value;
```

Comment:
Here I believe we should iterate on the entries and call the attachment constructor for each entry to get the same behaviour in cascade.